### PR TITLE
Adding ICs to set density from pressure and temperature

### LINKED ISSUE #15524: [fluid_properties]: Add initial conditions for density computed from pressure and temperature
## Reason

We want all applications using the fluid property module to be able to set density using the module, so they do not have reimplement the same code over and over.

## Design

This exists in THM and can be just moved into the module.

## Impact

Less code duplication in client apps, lower maintenance burden.

### PR DESCRIPTION
**Pull request by andrsd**
Closes #15524

### CODE REVIEW

**Comment by joshuahansel** [comment]
@andrsd In https://github.com/idaholab/moose/issues/14755 I added `p_sat` and `T_sat` as functions, so that they could be used either as ICs or PPs. Do you want to do that?

**Comment by joshuahansel** [comment]
Though I see you couple in variables instead of providing functions, so maybe you're not interested in the approach I suggested because then you'd need an additional step to translate your variables to functions...

File: modules/fluid_properties/doc/content/source/ics/RhoFromPressureTemperatureIC.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# RhoFromPressureTemperatureIC
+
+Initial condition to set density from given pressure and temperature using the
+`SinglePhaseFluidProperties` interface.
+
+!syntax description /ICs/RhoFromPressureTemperatureIC
```
Comment (joshuahansel) [inline modules/fluid_properties/doc/content/source/ics/RhoFromPressureTemperatureIC.md:6]: I'm thinking we should delete these lines if you provide a more detailed description, as you have here.

File: modules/fluid_properties/include/ics/RhoFromPressureTemperatureIC.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "InitialCondition.h"
+
+class SinglePhaseFluidProperties;
+
+/**
+ * Computes density from pressure and temperature
```
Comment (joshuahansel) [inline modules/fluid_properties/include/ics/RhoFromPressureTemperatureIC.h:17]: add "variables" at the end

File: modules/fluid_properties/include/ics/RhoVaporMixtureFromPressureTemperatureIC.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "InitialCondition.h"
+#include "VaporMixtureInterface.h"
+
+/**
+ * Computes the density of a vapor mixture from pressure and temperature.
```
Comment (joshuahansel) [inline modules/fluid_properties/include/ics/RhoVaporMixtureFromPressureTemperatureIC.h:16]: same

File: modules/fluid_properties/include/interfaces/VaporMixtureInterface.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
+#pragma once
```
Comment (joshuahansel) [inline modules/fluid_properties/include/interfaces/VaporMixtureInterface.h:1]: So is `VaporMixtureInterface` just a move from THM? Any changes? So there will need to be fixes to all the downstream apps?

File: modules/fluid_properties/test/tests/ics/rho_from_pressure_temperature/tests [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[Tests]
+  [./test]
```
Comment (joshuahansel) [inline modules/fluid_properties/test/tests/ics/rho_from_pressure_temperature/tests:2]: We should go ahead and remove dots in these files too

File: modules/fluid_properties/include/interfaces/VaporMixtureInterface.h (hunk 4 above)
Comment (joshuahansel) [reply modules/fluid_properties/include/interfaces/VaporMixtureInterface.h:1]: Hang on, all of this stuff came from THM? I was wondering why it seemed familiar...

**Review by joshuahansel — COMMENTED** [review]
Small questions and requested changes.

**Comment by andrsd** [comment]
> In #14755 I added `p_sat` and `T_sat` as functions, so that they could be used either as ICs or PPs. Do you want to do that?

I don't follow, I am moving stuff from THM so I can re-use it in the sub-channel code.

File: modules/fluid_properties/include/interfaces/VaporMixtureInterface.h (hunk 4 above)
Comment (andrsd) [reply modules/fluid_properties/include/interfaces/VaporMixtureInterface.h:1]: > So there will need to be fixes to all the downstream apps?

Yes...

File: modules/fluid_properties/test/tests/ics/rho_from_pressure_temperature/tests (hunk 5 above)
Comment (andrsd) [reply modules/fluid_properties/test/tests/ics/rho_from_pressure_temperature/tests:2]: ok

**Comment by joshuahansel** [comment]
> I don't follow, I am moving stuff from THM so I can re-use it in the sub-channel code.

I realized this later in the review. From the title, it sounded like it was new code. I see now that you mentioned moving it from THM in the corresponding issue. So ignore this comment.

File: modules/fluid_properties/test/tests/ics/rho_from_pressure_temperature/tests [hunk 6]
```diff
@@ -0,0 +1,8 @@
+[Tests]
+  [test]
+    type = 'CSVDiff'
+    input = 'test.i'
+    csvdiff = 'test_out.csv'
+    rel_err = 1e-10
```
Comment (joshuahansel) [inline modules/fluid_properties/test/tests/ics/rho_from_pressure_temperature/tests:6]: I'm sorry I missed this in the first review: we need requirements/design/issues for this and the other specs.

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: modules/fluid_properties/test/tests/ics/rho_from_pressure_temperature/tests (hunk 6 above)
Comment (andrsd) [reply modules/fluid_properties/test/tests/ics/rho_from_pressure_temperature/tests:6]: yeah, i thought there was a check for this in the pre-check recipe

**Comment by andrsd** [comment]
`VaporMixtureInterface` was missing the MOOSE header in the .h and .C file, so I added that as well.

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**